Fixes model property names not being serialized properly

nikcio opened the pull request:
When using `.AddNewtonsoftJson()` in our application we encountered that the models wasn't being serialized correctly from Skybrud.Umbraco.Redirects. This is because there was missing the `JsonProperty` attribute on the `RedirectModel`. Here `IsPermanent` and `ForwardQueryString` was serialized by those names when fetching the redirects and therefore the value was displayed incorrect when viewing a redirect in edit mode.

Therefore, I've added attributes to all models that was missing either `JsonProperty` or `JsonPropertyName` to preserve consistency in the models no matter what you're using. (System.Text.Json or Newtonsoft.Json).

I've tested this in our solution and it works with our setup.

Hope this helps @abjerner 😄 

Before:
![image](https://github.com/skybrud/Skybrud.Umbraco.Redirects/assets/24605285/2fb24e5b-c242-4d3a-984b-f179ca87a3e4)

After:
![image](https://github.com/skybrud/Skybrud.Umbraco.Redirects/assets/24605285/6a501183-3bb1-4740-a2c1-92df1d79eb15)

nikcio committed:
fix: Add attributes to all JSON models

abjerner commented:
Thanks 👍 

API controllers for the backoffice use **JSON.net** by default, so I've used `JsonResult` to serialize using **System.Text.Json** instead. It seems that `.AddNewtonsoftJson()` also changes the serializer used for `JsonResult`. which is a bit annoying - I didn't know that.

I guess there are different ways to go about fixing this, but adding attributes for both serializers like you've done for this PR seems like an easy fix. I need to find some time for testing this, but looks good to merge.

abjerner committed:
Introduced new "RedirectList" to replace anonymous class

#179 adds JSON attributes to properties in the "RedirectsSearchResultPagination" class, but this class isn't returned directly in the API. The existing implementation would return an anonymous class, which we can't add attributes to, so this commit introduces a "RedirectList" to be returned instead.

abjerner committed:
Fixed "backOfficeUrl" typo

The angular view for the redirects dashboard expects "backOfficeUrl" property in the returned JSON, but the "JsonProperty" attribute added in #179 would set this to "backofficeUrl".

abjerner commented:
@nikcio thanks again 👍

Vacation got in the way, but I've now had some time to test and merge this. Personally I'm not a fan of using `AddNewtonsoftJson` as it changes the serializer globally, but for people choosing to use it, this PR is a great fix.

I found a few minor issues, which fortunately was quick to fix. For one, the dashboard broke as the `pagination` would become `Pagination` when using `.AddNewtonsoftJson( options => { options.UseMemberCasing(); });`.
